build/bin/network-validator.go

```diff
@@ -106,18 +116,58 @@ func TestEndpoints(config reachabilityConfig) {
 	os.Exit(0)
 }
 
-func ValidateReachability(host string, port int, tlsDisabled bool) error {
+func getProxyCertificate(cacertFile string) (*x509.CertPool, error) {
+	if cacertFile == "" {
+		// default is being set.
+		return nil, nil
+	}
+
+	// Get the SystemCertPool, continue with an empty pool on error
+	rootCAs, _ := x509.SystemCertPool()
+	if rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	// Read in the cert file
+	cert, err := ioutil.ReadFile(cacertFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Append our cert to the system pool
+	if ok := rootCAs.AppendCertsFromPEM([]byte(cert)); !ok {
+		log.Println("No certs appended, using system certs only")
+	}
+
+	return rootCAs, nil
+}
+
+func ValidateReachability(host string, port int, tlsDisabled bool, cacertFile string) error {
 	var err error
 	endpoint := fmt.Sprintf("%s:%d", host, port)
 	httpClient := http.Client{
 		Timeout: *timeout,
 	}
 
-	// #nosec G402 -- Low chance of MITM, as the instance is short-lived, see OHSS-11465
-	if tlsDisabled {
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		}
+	// Setup Certs
+	rootCAs, err := getProxyCertificate(cacertFile)
+	if err != nil {
+		log.Fatalf("Failed to append %v to RootCAs: %v", rootCAs, err)
+		return err
+	}
+
+	// ProxyFromEnvironment enables reading configuration from env
+	// such as export HTTP_PROXY='http://us:pass@prox-server:8888' / export HTTPS_PROXY='http://us:pass@prox-server:8888'
+	// Insecure mod would be enabled if tlsDisabled was put as true. This is for dev purposes: e.g when the certificate is not known by certificate authorities.
+	httpClient.Transport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		TLSClientConfig: &tls.Config{
+			// #nosec G402 -- Low chance of MITM, as the instance is short-lived, see OHSS-11465
+			InsecureSkipVerify: tlsDisabled,
+			// RootCAs defines the set of root certificate authorities that clients use when verifying server certificates.
+			// If RootCAs is nil, TLS uses the host's root CA set.
```

Comment:
Is there documentation stating this? If so, mind dropping a link here? Just want to make sure that this is the default behavior.

Reply:
https://github.com/golang/go/blob/master/src/crypto/tls/common.go#L614